packages/vite/src/node/utils.ts

```diff
@@ -505,8 +505,11 @@ export function generateCodeFrame(
   start: number | Pos = 0,
   end?: number | Pos,
 ): string {
-  start = posToNumber(source, start)
-  end = end !== undefined ? posToNumber(source, end) : start
+  start = Math.max(posToNumber(source, start), 0)
+  end = Math.min(
+    end !== undefined ? posToNumber(source, end) : start,
+    source.length,
+  )
```

Comment:
https://github.com/vitejs/vite/blob/bcc4a6113521543ac17552c3b3cbda80514f0c04/packages/vite/src/node/plugins/css.ts#L1199-L1202
`{ endLine: 107, endColumn: 2 }` was passed here. So `posToNumber(source, { line: 107, column: 1 })` was called.
But the `source` only has 107 lines. So `posToNumber` returned a number greater than `source.length`.